script.js

```diff
@@ -1,60 +1,274 @@
+   arrRates.forEach(item => {
+      let row = document.createElement("tr");
+      let newNameRow = document.createElement("th");
+      newNameRow.scope = "row";
+      newNameRow.innerText = item.base;
+      row.appendChild(newNameRow);
+
+      addedColumns.forEach(rateKey => {
+         let newRate = document.createElement("td");
+         newRate.innerText = item.rates[rateKey] ? item.rates[rateKey].toFixed(5) : ''; // Fill with rate value or empty if not present
+         row.appendChild(newRate);
+      });
+
+      tableCurrencyRow.appendChild(row);
+   });
+}
+
+// Function to create the table
+function createTable() {
+   const tableCurrencyCol = document.querySelector('.table-col-currency');
+   const tableCurrencyRow = document.querySelector('.table-row-currency');
+
+   if (!tableCurrencyCol || !tableCurrencyRow) return;
+
+   const addedColumns = new Set();
+
+   createTableHeaders(tableCurrencyCol, addedColumns);
+   createTableRows(tableCurrencyRow, addedColumns);
+}
+
+createTable();
+
+// Function to populate select elements for currency rates
+function populateSelectElements(currencyElements, addedCurrencies, isBase = false) {
+   arrRates.forEach(item => {
+      const currencyKey = isBase ? [item.base] : Object.keys(item.rates);
+      currencyKey.forEach(key => {
+         if (!addedCurrencies.has(key)) {
+            addedCurrencies.add(key);
+            let newOption = document.createElement("option");
+            newOption.value = key;
+            newOption.innerText = key;
+            currencyElements.forEach(element => element.appendChild(newOption.cloneNode(true)));
+         }
+      });
+   });
+}
+
+// Function to create rate select elements
+function createRates() {
+   const selectCurrencyFrom = document.querySelectorAll('.select_currency_from');
+   const selectCurrencyTo = document.querySelectorAll('.select_currency_to');
+   if (!selectCurrencyFrom.length || !selectCurrencyTo.length) return;
+
+   const addedCurrenciesFrom = new Set();
+   const addedCurrenciesTo = new Set();
+
+   populateSelectElements(selectCurrencyFrom, addedCurrenciesFrom, true);
+   populateSelectElements(selectCurrencyTo, addedCurrenciesTo, false);
+}
+
+createRates();
+
+// Function to handle form submissions
+function handleFormSubmissions() {
+   document.querySelector('.formRate--js').addEventListener('submit', function(event) {
+      event.preventDefault();
+      addRate();
+      console.log(arrRates);
+   });
+
+   document.querySelector('.formUpdateRate--js').addEventListener('submit', function(event) {
+      event.preventDefault();
+      updateRate();
+      console.log(arrRates);
+   });
+
+   document.querySelector('.formConvert--js').addEventListener('input', function(event) {
+      event.preventDefault();
+      getConvert();
+   });
+
+   document.querySelector('.formSearchRate--js').addEventListener('submit', function(event) {
+      event.preventDefault();
+      searchRate();
+   });
+}
```

Comment:
Really great how you have compartmentalised your code in general into smaller functions